python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py

```diff
@@ -147,5 +147,5 @@ def build_model(self):
         if self.model_weights:
             model.set_weights(self.model_weights)
 
-        prediction = model(tensor).numpy().ravel()
-        return pd.DataFrame(prediction, columns=["predictions"])
+        prediction = list(model(tensor).numpy())
```

Comment:
cc @JiahaoYao 

hmmm, This doesn't work for general case, right? For example, Jimmy is outputting an image per prediction.
I thought yesterday I saw you using TensorArray?

Reply:
I was yesterday, but I thought we decided to use a list so there isn't a dependency on Ray Data?

In any case, I'd prefer to use `TensorArray`. I feel like it works a bit better.

Reply:
Another issue with the `list` approach is that we can't show outputs from `BatchPredictor.predict`.

```
outputs = batch_predictor.predict(
    data=test_dataset, feature_columns=["image"], unsqueeze=False
) 
outputs.show(1)  # Raises error during `repr`!
```

Reply:
@amogkam Are you ok with using TensorArray here and leave a TODO to factor out TensorArray to be just a util (independent of Ray Data)

Reply:
synced offline. Will go with list() for now and revisit after Clark is back.